packages/main/src/MenuItem.js

```diff
@@ -0,0 +1,173 @@
+		 *
+		 * @type {boolean}
+		 * @defaultvalue false
+		 * @public
+		 */
+		startsSection: {
+			type: Boolean,
+		},
+
+		/**
+		 * Defines whether <code>ui5-menu-item</code> is in disabled state.
+		 * <br><br>
+		 * <b>Note:</b> A disabled <code>ui5-menu-item</code> is noninteractive.
+		 * @type {boolean}
+		 * @defaultvalue false
+		 * @public
+		 */
+		disabled: {
+			type: Boolean,
+		},
+
+		/**
+		 * Indicates if the any of the element siblings have children items.
+		 * @type {boolean}
+		 * @private
+		 */
+		 _siblingsWithChildren: {
+			type: Boolean,
+			noAttribute: true,
+		},
+
+		/**
+		 * Indicates if the any of the element siblings have icon.
+		 * @type {boolean}
+		 * @private
+		 */
+		 _siblingsWithIcon: {
+			type: Boolean,
+			noAttribute: true,
+		},
+
+		/**
+		 * Stores Menu object with submenu items
+		 * @type {object}
+		 * @private
+		 */
+		 _subMenu: {
+			type: Object,
+		},
+
+		/**
+		 * Defines whether the submenu closing must be prevented
+		 * @type {boolean}
+		 * @private
+		 */
+		 _preventSubMenuClose: {
+			type: Boolean,
+			noAttribute: true,
+		},
+	},
+	managedSlots: true,
+	slots: /** @lends sap.ui.webcomponents.main.MenuItem.prototype */ {
+		/**
+		 * Defines the items of this component.
+		 *
+		 * @type {sap.ui.webcomponents.main.MenuItem[]}
+		 * @slot items
+		 * @public
+		 */
+		 "default": {
+			propertyName: "items",
+			type: HTMLElement,
+			invalidateOnChildChange: true,
+		},
+	},
+};
+
+/**
+ * @class
+ *
+ * <h3 class="comment-api-title">Overview</h3>
+ *
+ * <code>ui5-menu-item</code> is the item to use inside a <code>ui5-menu</code>.
+ * An arbitrary hierarchy structure can be represented by recursively nesting menu items.
+ *
+ * <h3>Usage</h3>
+ *
+ * <code>ui5-menu-item</code> is an abstract element, representing a node in a <code>ui5-menu</code>. The menu itself is rendered as a list,
+ * and each <code>ui5-menu-item</code> is represented by a list item (<code>ui5-li</code>) in that list. Therefore, you should only use
+ * <code>ui5-menu-item</code> directly in your apps. The <code>ui5-li</code> list item is internal for the list, and not intended for public use.
+ *
+ * For the <code>ui5-menu-item</code>
+ * <h3>ES6 Module Import</h3>
+ *
+ * <code>import @ui5/webcomponents/dist/MenuItem.js";</code>
+ *
+ * @constructor
+ * @author SAP SE
+ * @alias sap.ui.webcomponents.main.MenuItem
+ * @extends UI5Element
+ * @tagname ui5-menu-item
+ * @since 1.3.0
+ * @public
+ */
+class MenuItem extends UI5Element {
+	static get metadata() {
+		return metadata;
+	}
+
+	get hasChildren() {
+		return !!this.items.length;
+	}
+
+	get hasDummyIcon() {
+		return this._siblingsWithIcon && !this.icon;
+	}
+
+	get subMenuOpened() {
+		return !!Object.keys(this._subMenu).length;
+	}
+
+	/**
+	 * Returns the parent item of the current item.
+	 * @public
+	 * @returns {object} the item that opened the sub menu containing the current item.
+	 */
+	parent() {
```

Comment:
I would suggest not to provide such a function for the time being. First, I'm not sure about the name: if it is called just `parent` it should probably be a getter, not a method, similarly to how HMLT's native parentElement and parentNode work. Also, I'm not sure why people couldn't use `parentElement` recursively to traverse the HTML they created themselves. Finally, I think the app has other ways to get such information, if needed - for example if they have two "Save" manu items in different subtrees, they can give them different IDs if they need to know which one in particular was selected by the user.

Reply:
Done